pkg/network/usm/monitor_test.go

```diff
@@ -285,142 +344,209 @@ func (s *HTTPTestSuite) TestHTTPMonitorIntegrationWithNAT() {
-		}
+			requestsSum := 0
+			for key := range stats {
+				if key.Method == http.MethodUnknown {
+					t.Error("detected HTTP request with method unknown")
+				}
+				// we just want our requests
+				if strings.Contains(key.Path.Content, "/request-") &&
+					key.DstPort == 8080 &&
+					util.FromLowHigh(key.DstIPLow, key.DstIPHigh) == serverAddrIP {
+					requestsSum++
+				}
+			}
+
+			require.Equal(t, int64(0), dropped)
+			require.Equal(t, int64(0), rejected)
+			require.Equal(t, int64(0), malformed)
+			// requestGenerator() doesn't query 100 responses
+			require.Equal(t, int64(0), hits1XX)
+
+			require.Equal(t, int(100), requestsSum)
+		})
 	}
 }
 
 func (s *HTTPTestSuite) TestRSTPacketRegression() {
 	t := s.T()
-	monitor := newHTTPMonitor(t)
 
-	serverAddr := "127.0.0.1:8080"
-	srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
-		EnableKeepAlive: true,
-	})
-	t.Cleanup(srvDoneFn)
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
 
-	// Create a "raw" TCP socket that will serve as our HTTP client
-	// We do this in order to configure the socket option SO_LINGER
-	// so we can force a RST packet to be sent during termination
-	c, err := net.DialTimeout("tcp", serverAddr, 5*time.Second)
-	if err != nil {
-		t.Fatal(err)
-	}
+			monitor := newHTTPMonitor(t)
 
-	// Issue HTTP request
-	c.Write([]byte("GET /200/foobar HTTP/1.1\nHost: 127.0.0.1:8080\n\n"))
-	io.Copy(io.Discard, c)
+			serverAddr := "127.0.0.1:8080"
+			srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
+				EnableKeepAlive: true,
+			})
+			t.Cleanup(srvDoneFn)
 
-	// Configure SO_LINGER to 0 so that triggers an RST when the socket is terminated
-	require.NoError(t, c.(*net.TCPConn).SetLinger(0))
-	c.Close()
-	time.Sleep(100 * time.Millisecond)
+			// Create a "raw" TCP socket that will serve as our HTTP client
+			// We do this in order to configure the socket option SO_LINGER
+			// so we can force a RST packet to be sent during termination
+			c, err := net.DialTimeout("tcp", serverAddr, 5*time.Second)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-	// Assert that the HTTP request was correctly handled despite its forceful termination
-	stats := monitor.GetHTTPStats()
-	url, err := url.Parse("http://127.0.0.1:8080/200/foobar")
-	require.NoError(t, err)
-	includesRequest(t, stats, &nethttp.Request{URL: url})
+			// Issue HTTP request
+			c.Write([]byte("GET /200/foobar HTTP/1.1\nHost: 127.0.0.1:8080\n\n"))
+			io.Copy(io.Discard, c)
+
+			// Configure SO_LINGER to 0 so that triggers an RST when the socket is terminated
+			require.NoError(t, c.(*net.TCPConn).SetLinger(0))
+			c.Close()
+			time.Sleep(100 * time.Millisecond)
+
+			// Assert that // TODO: he HTTP request was correctly handled despite its forceful termination
```

Comment:
??

Reply:
don't know what happen here, seems the original was old line 357 https://github.com/DataDog/datadog-agent/pull/13311/files#diff-eba5a1cdf537b07c1956c9d5b65974f8fcf39b86d795e067ef1a868138d40597L357